v2/client.go

```diff
@@ -540,6 +540,12 @@ type client struct {
 	closed      chan struct{}
 }
 
+// Configuration to let the client knows which server it should connect to
+const (
+	ConnectToLowLatencyServer  = 0
+	ConnectToLowWorkLoadServer = 1
+)
+
```

Comment:
Maybe define a type for this? Also I think we can shorten the names since "ConnectTo" would be implied by the type name. And should we include "random" as the zero-value? For example:

```go
type SelectionCriteria int

const (
    Random SelectionCriteria = iota
    Latency
    Workload
)
```

Reply:
Make sense.

Though the Random seems to be a bit redundant as it is the current behavior already. However, I think explicit is better than implicit.